File: ShadowFrameworkLite20/src/shadow/renderer/data/SFTextureData.java
package shadow.renderer.data;

import java.util.StringTokenizer;

import shadow.image.SFRenderedTexturesSet;
import shadow.image.SFTexture;
import shadow.pipeline.SFPrimitive;
import shadow.system.data.SFDataCenterListener;
import shadow.system.data.SFInputStream;
import shadow.system.data.SFOutputStream;
import shadow.system.data.SFWritableDataObject;

public class SFTextureData extends SFPrimitive implements SFWritableDataObject{

	private int textureIndex;
	private SFLibraryReference<SFRenderedTexturesSet> reference=new SFLibraryReference<SFRenderedTexturesSet>();
	
	public SFTextureData() {
	}

	public void setTextureIndex(int textureIndex) {
		this.textureIndex = textureIndex;
	}

	public void setReference(String reference) {
		this.reference.setReference(reference);
	}

	public SFLibraryReference<SFRenderedTexturesSet> getReference() {
		return reference;
	}
	
	public SFTexture buildTextureReference() {
		
		final SFTexture texture=new SFTexture(null, textureIndex);
		
		getReference().retrieveDataset(new SFDataCenterListener<SFDataAsset<SFRenderedTexturesSet>>() {
			@Override
			public void onDatasetAvailable(String name, SFDataAsset<SFRenderedTexturesSet> dataset) {
				texture.setTexturesSet(dataset.getResource());
			}
		});
		
		return texture;
	}
	
	@Override
	public SFTextureData clone() {
		return new SFTextureData();
	}
	
	@Override
	public void readFromStream(SFInputStream stream) {
		textureIndex=stream.readByte();
		reference.setReference(stream.readName());
	}
	
	@Override
	public void writeOnStream(SFOutputStream stream) {
		stream.writeByte(textureIndex);
		stream.writeName(reference.getReference());
	}
	
	@Override
	public void setStringValue(String value) {
		StringTokenizer tok=new StringTokenizer(value,":");
		reference.setReference(tok.nextToken());
		if(tok.hasMoreTokens())
			textureIndex=new Integer(tok.nextToken());
	}
	@Override
	public String toStringValue() {
		return reference.getReference()+":";
	}
}

File: trunk/ShadowFrameworkJS/src/shadow/renderer/data/SFTextureData.js


function SFTextureData(){
	this.textureIndex = 0;
	this.reference=new SFLibraryReference();
}

inherit(SFTextureData,SFPrimitive);

SFTextureData.prototype["getReference"]=function(){
			return this.reference;
		};
		
SFTextureData.prototype["clone"]=function(){
			return new SFTextureData();
		};

SFTextureData.prototype["buildTextureReference"]=function(){
	
	var texture=new SFTexture(null, this.textureIndex);
	
	var dataset=this.getReference().retrieveDataset();
	if(dataset===undefined)
		throw "UNDEFINED";
	texture.setTexturesSet(dataset.getResource());
	
	return texture;
};

SFTextureData.prototype["readFromStream"]=function(stream){
	this.textureIndex=stream.readByte();
	this.reference.setReference(stream.readName());
};

SFTextureData.prototype["writeOnStream"]=function(stream){
	stream.writeByte(this.textureIndex);
	stream.writeName(this.reference.getReference());
};
